profiles/views.py

```diff
@@ -55,40 +54,26 @@ class SavedCompaniesCreate(CreateAPIView):
     serializer_class = SavedCompanySerializer
     pagination_class = ForumPagination
 
-    def post(self, request):
-        user = request.user
-        pk = request.data.get("company_pk")
-
-        # Check if the company is already in the user's saved list
-        if SavedCompany.objects.filter(user=user, company_id=pk).exists():
-            saved_company_destroyer = SavedCompaniesDestroy()
-            return saved_company_destroyer.destroy(request, pk)
-
-        serializer = SavedCompanySerializer(
-            data={"company": pk, "user": user.id}
-        )
-        if serializer.is_valid():
-            serializer.save()
-            return Response({"Company added": serializer.data})
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    def perform_create(self, serializer):
+        return serializer.save(user=self.request.user)
 
 
 class SavedCompaniesDestroy(DestroyAPIView):
     """
     Remove the company from the saved list.
     """
 
+    queryset = SavedCompany.objects.all()
     permission_classes = [IsAuthenticated]
 
-    def destroy(self, request, pk):
-        user = request.user
-        saved_company = get_object_or_404(
-            SavedCompany, company_id=pk, user=user
-        )
-        saved_company.delete()
-        return Response(
-            f"Company {pk} deleted", status=status.HTTP_204_NO_CONTENT
-        )
+    def get_object(self):
+        queryset = self.get_queryset()
+        lookup_kwargs = {
+            "user_id": self.request.user.id,
+            "company_id": self.kwargs["company_pk"],
+        }
+        obj = get_object_or_404(queryset, **lookup_kwargs)
+        return obj
```

Comment:
This is bad. DRF allows writing much cleaner (and shorter) code.

You don't have to override `get_object()` just to apply filtering by current user and translation of view kwarg `company_pk` into field name `company_id`.

The first issue is easily addressed by overriding `get_queryset()` (instead of using `queryset` property), the second — by using `lookup_field` and `lookup_url_kwarg` (please read documentation on those).